daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/NonBaseTenTextNumberParser.scala

```diff
@@ -65,13 +65,14 @@ class ConvertNonBaseTenTextNumberParser(
         return
     }
 
-    if (!primNumeric.isValidRange(bi)) {
-      PE(state, "Parsed %s is out of range for the type: %s",
-        context.optPrimType.get.globalQName, bi.toString)
-      return
+    val num = try {
+      primNumeric.fromNumber(bi)
+    } catch {
+      case e: Throwable => {
```

Comment:
All these catch of Throwable have to change. Basically you can never catch Throwable unless you are some top-level application structure (E.g., our CLI might have an excuse to catch throwable)

If you catch throwable, then things like Java out-of-memory problems will get masked.  And in this case, captured, turned into processing errors, and they then cause backtracking..... not good.